modular_skyrat/modules/self_actualization_device/code/self_actualization_device.dm

```diff
@@ -115,49 +125,112 @@
-		open_machine()
-		playsound(src, 'sound/machines/microwave/microwave-end.ogg', 100, FALSE)
 
-/obj/machinery/self_actualization_device/screwdriver_act(mob/living/user, obj/item/I)
+	if(!ishuman(occupant))
+		return FALSE
+
+	var/mob/living/carbon/human/patient = occupant
+	var/original_name = patient.dna.real_name
+
+	//Organ damage saving code.
+	var/heart_damage = check_organ(patient, /obj/item/organ/heart)
+	var/liver_damage = check_organ(patient, /obj/item/organ/liver)
+	var/lung_damage = check_organ(patient, /obj/item/organ/lungs)
+	var/stomach_damage = check_organ(patient, /obj/item/organ/stomach)
+	var/appendix_damage = check_organ(patient, /obj/item/organ/appendix)
+	var/brain_damage = check_organ(patient, /obj/item/organ/brain)
+	var/eye_damage = check_organ(patient, /obj/item/organ/eyes)
+	var/ear_damage = check_organ(patient, /obj/item/organ/ears)
+
+	var/list/trauma_list = list()
+	if(patient.get_traumas())
+		for(var/datum/brain_trauma/trauma in patient.get_traumas())
+			trauma_list += trauma
+
+	var/brute_damage = patient.getBruteLoss()
+	var/burn_damage = patient.getFireLoss()
+
+	if(obj_flags & EMAGGED)
+		patient.monkeyize()
+
+	else
+		patient?.client?.prefs?.safe_transfer_prefs_to(patient)
+		patient.dna.update_dna_identity()
+		log_game("[key_name(patient)] used a Self-Actualization Device at [loc_name(src)].")
+
+		if(patient.dna.real_name != original_name)
+			message_admins("[key_name_admin(patient)] has used the Self-Actualization Device, and changed the name of their character. \
+			Original Name: [original_name], New Name: [patient.dna.real_name]. \
+			This may be a false positive from changing from a humanized monkey into a character, so be careful.")
+
+	// Apply organ damage
+	patient.adjustOrganLoss(ORGAN_SLOT_HEART, heart_damage)
+	patient.adjustOrganLoss(ORGAN_SLOT_LIVER, liver_damage)
+	patient.adjustOrganLoss(ORGAN_SLOT_LUNGS, lung_damage)
+	patient.adjustOrganLoss(ORGAN_SLOT_STOMACH, stomach_damage)
+	patient.adjustOrganLoss(ORGAN_SLOT_APPENDIX, appendix_damage)
+	// Head organ damage.
+	patient.adjustOrganLoss(ORGAN_SLOT_EYES, eye_damage)
+	patient.adjustOrganLoss(ORGAN_SLOT_EARS, ear_damage)
+	patient.adjustOrganLoss(ORGAN_SLOT_BRAIN, brain_damage)
+
+	//Re-Applies Trauma
+	var/obj/item/organ/brain/patient_brain = patient.getorgan(/obj/item/organ/brain)
+
+	if(length(trauma_list))
+		patient_brain.traumas = trauma_list
+
+	//Re-Applies Damage
+	patient.adjustBruteLoss(brute_damage)
+	patient.adjustFireLoss(burn_damage)
+
+	open_machine()
+	playsound(src, 'sound/machines/microwave/microwave-end.ogg', 100, FALSE)
+
+/// Checks the damage on the inputed organ and stores it.
+/obj/machinery/self_actualization_device/proc/check_organ(mob/living/carbon/human/patient, obj/item/organ/organ_to_check)
+	var/obj/item/organ/organ_to_track = patient.getorgan(organ_to_check)
+
+	// If the organ is missing, the organ damage is automatically set to 100.
+	if(!organ_to_track)
```

Comment:
Do we really want to give them a new bout of appendicitis if they've had it earlier in the round? I think there's an argument to be made either way, but I'd lean towards no.